src/fileActions.jsx

```diff
@@ -625,3 +625,31 @@ export const updateFile = (file, currentPath) => {
                     return file;
             }, exc => console.error(`Adding file ${file} failed: ${exc.toString()}`));
 };
+
+export const downloadFile = (currentPath, selected) => {
+    const query = window.btoa(JSON.stringify({
+        payload: "fsread1",
+        binary: "raw",
+        path: `${currentPath}/${selected.name}`,
+        superuser: "try",
+        // TODO: is this a good value?
+        max_read_size: 150 * 1024 * 1024,
+        external: {
+            "content-disposition": `attachment; filename=\\"${selected.name}\\"`,
+            "content-type": "application/octet-stream"
+        }
+    }));
+
+    const prefix = (new URL(cockpit.transport.uri("channel/" + cockpit.transport.csrf_token))).pathname;
+    const url = prefix + "?" + query;
+    return new Promise((resolve, reject) => {
+        const iframe = document.createElement("iframe");
+        iframe.setAttribute("src", url);
+        iframe.setAttribute("hidden", "hidden");
+        iframe.addEventListener("load", () => {
```

Comment:
What happens if the download gets interrupted or this fails due to permission errors etc.? I don't see any error handling from the fsread1 channel here.

Reply:
Right, I guess you just get a "download filed error" from the browser. Setting an alert will be difficult I think.

Reply:
That's fine -- if this appears as a proper download, then the browser should handle everything.

Reply:
... on the browser side. But on the bridge side we probably still need some error handling so that this `Promise` rejects or resolves eventually?

Reply:
Check if the load event reports a 403 back or anything not 200 so we can do error handling.

Reply:
The load event gives us zero information if the url was loaded, neither does onError. I fear we have no way of detection this. One idea is to call cockpit.file.spawn('head -n 4').then(() => { iframe code }).